Compilation issues with PHP master (8,6-dev)

<!--
You are about to open an issue for the MongoDB PHP extension. If you instead
intended to report an issue with the MongoDB PHP library, please do so via its
repository[^1].

If you've identified a security vulnerability in a driver or any other MongoDB
project, please create a vulnerability report[^2].

[^1]: https://github.com/mongodb/mongo-php-library
[^2]: https://docs.mongodb.org/manual/tutorial/create-a-vulnerability-report
-->

### Bug Report

Compilation with php-master failes with:

```
Error Output:
================
mkdir: cannot create directory 'src/libmongocrypt/src/crypto/.libs': File exists
mkdir: cannot create directory 'src/libmongocrypt/src/os_win/.libs': File exists
mkdir: cannot create directory 'src/MongoDB/Monitoring/.libs': File exists
mkdir: cannot create directory 'src/MongoDB/Monitoring/.libs': File exists
/home/derick/.config/pie/php8.6_b3a3bdfe9f58cd5449048cea5d2d231b/vendor/mongodb/mongodb-extension/mongodb-2.1.4/src/MongoDB/Cursor.c: In function ‘zim_MongoDB_Driver_Cursor_toArray’:
/home/derick/.config/pie/php8.6_b3a3bdfe9f58cd5449048cea5d2d231b/vendor/mongodb/mongodb-extension/mongodb-2.1.4/src/MongoDB/Cursor.c:126:17: error: implicit declaration of function ‘zval_dtor’ [-Wimplicit-function-declaration]
  126 |                 zval_dtor(return_value);
      |                 ^~~~~~~~~
/home/derick/.config/pie/php8.6_b3a3bdfe9f58cd5449048cea5d2d231b/vendor/mongodb/mongodb-extension/mongodb-2.1.4/src/MongoDB/ServerApi.c: In function ‘php_phongo_serverapi_init_from_hash’:
/home/derick/.config/pie/php8.6_b3a3bdfe9f58cd5449048cea5d2d231b/vendor/mongodb/mongodb-extension/mongodb-2.1.4/src/MongoDB/ServerApi.c:70:75: error: implicit declaration of function ‘ZVAL_IS_NULL’; did you mean ‘ZVAL_NULL’? [-Wimplicit-function-declaration]
   70 |         if ((strict = zend_hash_str_find(props, ZEND_STRL("strict"))) && !ZVAL_IS_NULL(strict)) {
      |                                                                           ^~~~~~~~~~~~
      |                                                                           ZVAL_NULL
/home/derick/.config/pie/php8.6_b3a3bdfe9f58cd5449048cea5d2d231b/vendor/mongodb/mongodb-extension/mongodb-2.1.4/src/MongoDB/ServerApi.c:88:27: error: implicit declaration of function ‘zval_is_true’; did you mean ‘zend_is_true’? [-Wimplicit-function-declaration]
   88 |                 strict && zval_is_true(strict),
      |                           ^~~~~~~~~~~~
      |                           zend_is_true
make: *** [Makefile:1152: src/MongoDB/Cursor.lo] Error 1
make: *** Waiting for unfinished jobs....
make: *** [Makefile:1173: src/MongoDB/ServerApi.lo] Error 1
```

- `zval_dtor()` needs to be replaces by `zval_ptr_dtor_nogc()`
- `ZVAL_IS_NULL` needs to be replaced with `Z_ISNULL()`
- `zval_is_true()` needs to be replaced wutg `zend_is_true()`

See: https://github.com/php/php-src/blob/master/UPGRADING.INTERNALS#L17

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -14,6 +14,7 @@ jobs:
   linux-test:
     name: "Linux Tests"
     runs-on: "${{ matrix.os }}"
+    continue-on-error: ${{ matrix.allow_failures || false }}
 
     strategy:
       fail-fast: true
@@ -47,6 +48,11 @@ jobs:
             php-version: "8.1"
             mongodb-version: "6.0"
             topology: "sharded_cluster"
+          - os: "ubuntu-24.04"
+            php-version: "8.6"
+            mongodb-version: "8.0"
+            topology: "replica_set"
+            allow_failures: true
 
     steps:
       - name: "Checkout"

--- a/src/MongoDB/Cursor.c
+++ b/src/MongoDB/Cursor.c
@@ -123,7 +123,7 @@ static PHP_METHOD(MongoDB_Driver_Cursor, toArray)
 	array_init(return_value);
 
 	if (spl_iterator_apply(getThis(), php_phongo_cursor_to_array_apply, (void*) return_value) != SUCCESS) {
-		zval_dtor(return_value);
+		zval_ptr_dtor_nogc(return_value);
 		RETURN_NULL();
 	}
 }

--- a/src/MongoDB/ServerApi.c
+++ b/src/MongoDB/ServerApi.c
@@ -67,14 +67,14 @@ static bool php_phongo_serverapi_init_from_hash(php_phongo_serverapi_t* intern, 
 		return false;
 	}
 
-	if ((strict = zend_hash_str_find(props, ZEND_STRL("strict"))) && !ZVAL_IS_NULL(strict)) {
+	if ((strict = zend_hash_str_find(props, ZEND_STRL("strict"))) && !Z_ISNULL_P(strict)) {
 		if (Z_TYPE_P(strict) != IS_TRUE && Z_TYPE_P(strict) != IS_FALSE) {
 			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "%s initialization requires \"strict\" field to be bool or null", ZSTR_VAL(php_phongo_serverapi_ce->name));
 			return false;
 		}
 	}
 
-	if ((deprecation_errors = zend_hash_str_find(props, ZEND_STRL("deprecationErrors"))) && !ZVAL_IS_NULL(deprecation_errors)) {
+	if ((deprecation_errors = zend_hash_str_find(props, ZEND_STRL("deprecationErrors"))) && !Z_ISNULL_P(deprecation_errors)) {
 		if (Z_TYPE_P(deprecation_errors) != IS_TRUE && Z_TYPE_P(deprecation_errors) != IS_FALSE) {
 			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "%s initialization requires \"deprecationErrors\" field to be bool or null", ZSTR_VAL(php_phongo_serverapi_ce->name));
 			return false;
@@ -84,10 +84,10 @@ static bool php_phongo_serverapi_init_from_hash(php_phongo_serverapi_t* intern, 
 	return php_phongo_serverapi_create_libmongoc_object(
 		&intern->server_api,
 		Z_STR_P(version),
-		strict && !ZVAL_IS_NULL(strict),
-		strict && zval_is_true(strict),
-		deprecation_errors && !ZVAL_IS_NULL(deprecation_errors),
-		deprecation_errors && zval_is_true(deprecation_errors));
+		strict && !Z_ISNULL_P(strict),
+		strict && zend_is_true(strict),
+		deprecation_errors && !Z_ISNULL_P(deprecation_errors),
+		deprecation_errors && zend_is_true(deprecation_errors));
 }
 
 /* Constructs a new ServerApi object */

